src/io/files.js

```diff
@@ -1093,8 +1158,72 @@ p5.prototype.httpDo = function() {
         'Content-Type': contentType
       })
     });
+    // do some sort of smart type checking
+    if (!type) {
+      if (path.indexOf('json') !== -1) {
+        type = 'json';
+      } else if (path.indexOf('xml') !== -1) {
+        type = 'xml';
+      } else {
+        type = 'text';
+      }
+    }
+
+    if (type === 'jsonp') {
+      promise = fetchJsonp(path, jsonpOptions);
+    } else {
+      promise = fetch(request);
+    }
+    promise = promise.then(function(res) {
```

Comment:
this line feels weird to me, can you explain what's going on here?

Reply:
This is the same code as present before, I'll add another parameter to the getFile() method so the code isn't repeated twice.